mllib/src/main/scala/org/apache/spark/ml/tuning/ParamGridBuilder.scala

```diff
@@ -100,13 +100,16 @@ class ParamGridBuilder {
    * Builds and returns all combinations of parameters specified by the param grid.
    */
   def build(): Array[ParamMap] = {
-    var paramMaps = Array(new ParamMap)
-    paramGrid.foreach { case (param, values) =>
-      val newParamMaps = values.flatMap { v =>
-        paramMaps.map(_.copy.put(param.asInstanceOf[Param[Any]], v))
+    if (paramGrid.isEmpty) Array.empty[ParamMap]
```

Comment:
style: Can you please put braces around this part as well to match the braces around the "else" part?  (That comment from @srowen was hidden by the update)